### PR TITLE
Fixing NULL deref and a race condition

### PR DESCRIPTION
**Pull request by MushMal**
We had a regression with NULL deref as the remote peer connection was not being removed from the hash table. This is what's been causing the crashes on some reported cases.

There is also a potential for a race condition that's also been fixed with these changes.

Issue #801 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/Common.c [hunk 1]
```diff
@@ -868,16 +864,18 @@ STATUS sessionCleanupWait(PSampleConfiguration pSampleConfiguration)
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
     PSampleStreamingSession pSampleStreamingSession = NULL;
-    UINT32 i;
-    BOOL locked = FALSE;
+    UINT32 i, clientIdHash;
+    ;
```
Comment (hassanctech) [inline samples/Common.c:868]: Stray `;`

File: samples/Common.c [hunk 2]
```diff
@@ -890,15 +888,19 @@ STATUS sessionCleanupWait(PSampleConfiguration pSampleConfiguration)
                 pSampleConfiguration->sampleStreamingSessionList[i] =
                     pSampleConfiguration->sampleStreamingSessionList[pSampleConfiguration->streamingSessionCount];
 
+                // Remove from the hash table
+                clientIdHash = COMPUTE_CRC32((PBYTE) pSampleStreamingSession->peerId, (UINT32) STRLEN(pSampleStreamingSession->peerId));
+                CHK_STATUS(hashTableContains(pSampleConfiguration->pRtcPeerConnectionForRemoteClient, clientIdHash, &peerConnectionFound));
+                if (peerConnectionFound) {
+                    CHK_STATUS(hashTableRemove(pSampleConfiguration->pRtcPeerConnectionForRemoteClient, clientIdHash));
```
Comment (disa6302) [inline samples/Common.c:895]: Do we want to to jump to CleanUp if `hashTableRemove` fails? We would not be freeing the streaming session in this case.

File: samples/Common.c (hunk 1 above)
Comment (MushMal) [reply samples/Common.c:868]: D'oh!. Will fix. Apparently, the clang formatter does smart things :)

File: samples/Common.c (hunk 2 above)
Comment (MushMal) [reply samples/Common.c:895]: This is the case when we have something unexpected. I think the best would be to fail indeed.

**Review by hassanctech — APPROVED** [review]
LGTM

**Review by disa6302 — APPROVED** [review]

**Pull request merged**